pkg/resources/gateway/configmap.go

```diff
@@ -90,6 +111,18 @@ func (r *Reconciler) configMap() (resources.Resource, string, error) {
 				ClientKey:  "/run/etcd/certs/client/tls.key",
 			},
 		}
+	case cfgv1beta1.StorageTypeJetStream:
+		nats := &opnicorev1beta1.NatsCluster{}
+		if err := r.client.Get(r.ctx, types.NamespacedName{
+			Namespace: r.gw.Namespace,
+			Name:      r.gw.Spec.NatsRef.Name,
+		}, nats); err != nil {
+			return nil, "", err
+		}
+		gatewayConf.Spec.Storage.JetStream = &cfgv1beta1.JetStreamStorageSpec{
+			Endpoint:     fmt.Sprintf("nats://%s-nats-client.%s.svc:%d", nats.Name, nats.Namespace, natsutil.DefaultClientPort),
+			NkeySeedPath: fmt.Sprintf("%s/seed", natsutil.NkeyDir),
+		}
```

Comment:
I feel like we could probably reuse https://github.com/rancher/opni/blob/main/pkg/util/nats/nats.go#L28 here